packages/rest/src/resource.ts

```diff
@@ -97,10 +97,11 @@ function cloneKeyProperties(context: DecoratorContext, target: Model, resourceTy
     const { keyProperty } = resourceKey;
     const keyName = getKeyName(program, keyProperty);
 
-    // Filter out the @visibility decorator because it might affect metadata
-    // filtering
     const decorators = [
-      ...keyProperty.decorators.filter((d) => d.decorator !== $visibility),
+      // Filter out the @visibility decorator because it might affect metadata
+      // filtering. NOTE: Check for name equality instead of function equality
+      // to deal with multiple copies of core being used.
+      ...keyProperty.decorators.filter((d) => d.decorator.name !== $visibility.name),
```

Comment:
This is dirty and if someone had `MyNamespace.visibility`, we'd filter that too. I'm not sure of the best way to fix this or if this is maybe acceptable. cc @bterlson

Reply:
Is there a more thorough test for "equivalence" we could apply here without falling back to name but without relying on strict equality of the function?

Reply:
I couldn't find one that already existed. I thought about plumbing through the decorator symbol onto the DecoratorApplication and comparing getFullyQualifiedName. Somewhat expensive. I also thought about storing a symbol on $visibility function and exposing a isVisibilityDecorator from core but this pollutes public API of core and would only work when all versions of core in play are updated with the fix.